### PR TITLE
Feature/autofocus

### PR DESCRIPTION
**Pull request by roesslerj**
- [x] the code follows the [Clean Code](https://clean-code-developer.com/) guidelines.
- [x] the necessary tests are either created or updated.
- [x] all status checks (Travis CI, SonarCloud, etc.) pass.
- [x] your commit history is cleaned up.
- [x] you updated the changelog.
- [ ] you updated necessary documentation within [retest/docs](https://github.com/retest/docs).

### CODE REVIEW

File: src/main/resources/javascript/getAllElementsByPath.js [hunk 1]
```diff
@@ -282,6 +282,19 @@ function isDisabled(node) {
     return node.disabled ? true : false;
 }
 
+function hasAutofocus(node) {
```
Comment (diba1013) [inline src/main/resources/javascript/getAllElementsByPath.js:285]: Just to drop in: @githubert What is the state of the recheck-web-js? Shouldn't this be done there`?

File: CHANGELOG.md [hunk 2]
```diff
@@ -35,6 +35,8 @@ Table of Contents
 
 ### Improvements
 
+* Add `autofocus` as attribute (needs special handling).
```
Comment (diba1013) [inline CHANGELOG.md:38]: That would be a breaking change, wouldn't it?

File: src/main/resources/javascript/getAllElementsByPath.js (hunk 1 above)
Comment (roesslerj) [reply src/main/resources/javascript/getAllElementsByPath.js:285]: As long as this hasn't been finished, we need to keep them in sync. But from my understanding, this is a minimal change in the JS.

File: CHANGELOG.md (hunk 2 above)
Comment (roesslerj) [reply CHANGELOG.md:38]: Iff an HTML contains autofocus, that change needs to be accepted. It surely isn't API breaking.

Comment (diba1013) [reply CHANGELOG.md:38]: > Breaking Changes when a change requires manual activity at update.

This is the case. While it is not an API change, it requires user activity. See the similarity to tabindex in the unrealeased section.

File: src/main/resources/javascript/getAllElementsByPath.js (hunk 1 above)
Comment (diba1013) [reply src/main/resources/javascript/getAllElementsByPath.js:285]: Then I would suggest that you create an corresponding PR in recheck-web-js? Just that this is not lost when transitioning. 

Just to clarify: I do not know what the best approach is here. In my understanding, this was already in progress. But let's wait what @githubert thinks.

Comment (githubert) [reply src/main/resources/javascript/getAllElementsByPath.js:285]: An additional PR would be nice, so I can implement it over there. Once I figure out how to make the Internet Explorer happy, it'll be ready. (Last status was that it successfully runs the script and extracts the attributes, but then it goes wrong _somewhere_ in the whole process)

Comment (githubert) [reply src/main/resources/javascript/getAllElementsByPath.js:285]: Or no PR, but open an issue.

Comment (roesslerj) [reply src/main/resources/javascript/getAllElementsByPath.js:285]: Done: https://github.com/retest/recheck-web-js/issues/61

**Pull request merged**